mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/ClassLoaderHandle.java

```diff
@@ -64,4 +65,12 @@ public void close() throws IOException {
             }
         };
     }
+
+    static UserCodeClassLoader createUserCodeClassloader(ExecuteStageRequest executeStageRequest, ClassLoaderHandle classLoaderHandle) throws Exception {
+        // triggers the download of all missing jar files from the job manager
+        final UserCodeClassLoader userCodeClassLoader =
+            classLoaderHandle.getOrResolveClassLoader(ImmutableList.of(executeStageRequest.getJobJarUrl().toURI()), ImmutableList.of());
+
+        return userCodeClassLoader;
+    }
```

Comment:
Instead of having this as a static method, you can have it as a default method of the ClassLoaderHandle interface. This way, it's a method of the class rather than a utility which is what a static method is. 

```
default UserCodeClassLoader createUserCodeClassLoader(ExecuteStageRequest executeStageRequest)
```